tests/test_cmd2.py

```diff
@@ -504,25 +505,60 @@ def test_save_invalid_path(base_app, capsys):
 
 
 def test_output_redirection(base_app):
-    # TODO: Use a temporary directory/file for this file
-    filename = 'out.txt'
-
-    # Verify that writing to a file works
-    run_cmd(base_app, 'help > {}'.format(filename))
-    expected = normalize(BASE_HELP)
-    with open(filename) as f:
-        content = normalize(f.read())
-    assert content == expected
-
-    # Verify that appending to a file also works
-    run_cmd(base_app, 'help history >> {}'.format(filename))
-    expected = normalize(BASE_HELP + '\n' + HELP_HISTORY)
-    with open(filename) as f:
-        content = normalize(f.read())
-    assert content == expected
-
-    # Delete file that was created
-    os.remove(filename)
+    fd, filename = tempfile.mkstemp(prefix='cmd2_test', suffix='.txt')
+    
+    try:
+        # Verify that writing to a file works
+        run_cmd(base_app, 'help > {}'.format(filename))
+        expected = normalize(BASE_HELP)
+        with open(filename) as f:
+            content = normalize(f.read())
+        assert content == expected
+
+        # Verify that appending to a file also works
+        run_cmd(base_app, 'help history >> {}'.format(filename))
+        expected = normalize(BASE_HELP + '\n' + HELP_HISTORY)
+        with open(filename) as f:
+            content = normalize(f.read())
+        assert content == expected
+    except:
+        raise
+    finally:
+        os.remove(filename)
```

Comment:
Something is going wrong with these unit test changes on Windows in both Python 2.7 and Python 3.x.  The code looks OK to me.  I'll boot up my Windows VM and see if I can get some insight into what is happening.